### DISCUSSION TITLE
How can I know if published or not in the iOS client?

### DISCUSSION THREAD

**Discussion by austinwg** [rabbitmq/rabbitmq-objc-client#223] [Q&A] [answered]
Hey! Can you please let me know if the iOS SDK can tell if a message has been published/delivery or not?
I used the `basicPublish` function, but there is no callback to return the published result.
In the Android SDK, the result was known through an exception.
Is there a similar mechanism in the iOS SDK?

**Comment by lukebakken** [comment]
I'm sure there is a way to use publisher confirmations with this library.

Java example - https://www.rabbitmq.com/federation-reference.html#upstreams

**Reply by austinwg** [reply]
Hey @lukebakken Thanks for your reply! But I want to get example iOS code.

**Reply by michaelklishin** [reply, marked as answer]
You are welcome to take a look at the tests, e.g.

 * [one](https://github.com/rabbitmq/rabbitmq-objc-client/blob/30c467b41212235f4cca78c4431a12683b68f91d/RMQClientIntegrationTests/ConsumersIntegrationTest.swift#L356-L383)
 * [two](https://github.com/rabbitmq/rabbitmq-objc-client/blob/30c467b41212235f4cca78c4431a12683b68f91d/RMQClientTests/PublisherConfirmationTest.swift#L45)
